subsys/net/l2/ethernet/gptp/gptp_mi.c

```diff
@@ -837,13 +835,17 @@ static void gptp_update_local_port_clock(void)
 	skip_clock_set:
 		irq_unlock(key);
 	} else {
-		if (nanosecond_diff < -200) {
-			nanosecond_diff = -200;
-		} else if (nanosecond_diff > 200) {
-			nanosecond_diff = 200;
-		}
+		double ppb = gptp_servo_pi(nanosecond_diff);
 
-		ptp_clock_adjust(clk, nanosecond_diff);
+		ptp_clock_rate_adjust(clk, 1.0 + (ppb / 1000000000.0));
+		/* This prints continuous messages but can be enabled to see
+		 * the real time convergence of synchronization offset.
+		 * This is what users care about.
+		 */
+		if (0) {
```

Comment:
I suggest creating a Kconfig symbol for this so that user does not need to adjust the source code to see the prints.
This could be named for example `CONFIG_NET_GPTP_MONITOR_SYNC_DIFF` or similar.
Then you could do here
```
if (IS_ENABLED(CONFIG_NET_GPTP_MONITOR_SYNC_DIFF)) {
....
}
```

Reply:
Thanks a lot, will do that.